kotlintest-core/src/main/kotlin/io/kotlintest/specs/AbstractBehaviorSpec.kt

```diff
@@ -12,22 +12,22 @@ abstract class AbstractBehaviorSpec(body: AbstractBehaviorSpec.() -> Unit = {})
 
   @KotlinTestDsl
   inner class GivenContext(val context: TestContext) {
-    fun And(name: String, test: WhenContext.() -> Unit) = and(name, test)
-    fun and(name: String, test: WhenContext.() -> Unit) = add("Add: $name", test)
-    fun When(name: String, test: WhenContext.() -> Unit) = `when`(name, test)
-    fun `when`(name: String, test: WhenContext.() -> Unit) = add("When: $name", test)
-    private fun add(name: String, test: WhenContext.() -> Unit) =
+    fun And(name: String, test: suspend WhenContext.() -> Unit) = and(name, test)
```

Comment:
I don't know what is better, however possible right solution is duplicate functions: with corotines support and without it